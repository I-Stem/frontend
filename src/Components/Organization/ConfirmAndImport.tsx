import React, { useEffect, useRef, useState } from "react";
import { NextPage } from "next";
import Head from "next/head";
import { connect } from "react-redux";
import { useRouter } from "next/router";
// #endregion Global Imports

// #region Local Imports
import "./style.scss";
import { IStemServices, IStore } from "@Interfaces";
import { Wrapper } from "@Components";
import { FormLayout } from "@Components/HOC/Dashboard/FormHOC";
import { useAppAbility } from "src/Hooks/useAppAbility";
import { BlueButton } from "@Components/HOC/Dashboard";
import { Table } from "react-bootstrap";
import { UniversityPortal } from "@Services";
import { STUDENTS } from "@Definitions/Constants/universityRoutes";
import { DialogMessageBox } from "@Components/Basic/Dialog";
import { UserType } from "@Definitions/Constants";
import { UniversityPortalActions } from "src/Actions/UniversityActions";
import { getInvitationResponseMessage } from "@Services/helper/utils";
import Pagination from "@Components/HOC/Pagination";
import Error from "next/error";

const ConfirmImportComponent: React.FC = (props: any) => {
  const initialFocus = useRef<HTMLDivElement>(null);
  const router = useRouter();
  const { can } = useAppAbility();
  const access = can("VIEW", "STUDENTS");
  const studentData = props.csvFile;

  const [currentPage, setCurrentPage] = useState(1);
  const indexLastStud = currentPage * 10;
  const indexFirstStud = indexLastStud - 10;
  const currentStudents = studentData.slice(indexFirstStud, indexLastStud);
  const [messageBox, setMessageBox] = useState(false);
  const [dialogMessage, setDialogMessage] = useState("");
  const [dialogHeading, setDialogHeading] = useState("");
  const { organizationCode, userType } = props.user;
  useEffect(() => {
    initialFocus?.current?.focus();
  });
  const rows = currentStudents?.map((data: any) => {
    return (
      <tr key={data.EMAIL}>
        <td>{data.NAME}</td>
        <td>{data.EMAIL}</td>
        <td>{data.ROLL_NO}</td>
      </tr>
    );
  });
  const handlePageNumber = (val: any) => {
    setCurrentPage(Number(val));
  };
  const toggleMessageModal = () => {
    setMessageBox(!messageBox);
  };

  const handleNextPage = () => {
    setCurrentPage(currentPage + 1);
  };
  const handlePreviousPage = () => {
    setCurrentPage(currentPage - 1);
  };
  const fullNames = Object.values(props.csvFile).map((data: any) =>
    String(data.NAME)
  );
  const emails = Object.values(props.csvFile).map((data: any) =>
    String(data.EMAIL)
  );

  const rollNo = Object.values(props.csvFile).map((data: any) =>
    String(data.ROLL_NO)
  );
  const handleConfirm = () => {
    UniversityPortal.studentInvite({
      fullNames,
      emails,
      organization: organizationCode,
      rollNos: rollNo,
      role: "STUDENT",
    })
      .then((results: any) => {
        if (results.code === 200) {
          const message = getInvitationResponseMessage(
            results.data.newUsers,
            results.data.existingUsers
          );
          setDialogMessage(message);
          setDialogHeading("Invitation Success");
          setMessageBox(!messageBox);
          props.resetCsv();
        }
      })
      .catch((err: any) => {
        setDialogMessage(
          "Error occurred while sending invitations through emails. Try Again!"
        );
        setDialogHeading("Invitation Failed");
        setMessageBox(!messageBox);
      });
  };

  return access ? (
    <Wrapper>
      <Head>
        <title>Confirm and Import | I-Stem</title>
      </Head>
      <FormLayout form="confirmAndMerge" hideFooter={true}>
        <div className="lip-margin">
          <div tabIndex={-1} ref={initialFocus}>
            <h2 className="mt-8 lip-title">
              STEP 3: CONFIRM AND IMPORT ({studentData.length}{" "}
              {userType === UserType.BUSINESS ? "EMPLOYEES" : "STUDENTS"})
            </h2>
          </div>
          <div className="">
            <Table style={{ marginTop: "1rem" }}>
              <thead>
                <tr style={{ borderTop: "hidden" }}>
                  <th>NAME</th>
                  <th>Email</th>
                  <th>ROLL NUMBER</th>
                </tr>
              </thead>
              <tbody>{rows}</tbody>
            </Table>
          </div>
          <Pagination
            totalItems={studentData.length}
            currentPage={currentPage}
            handleNextPage={handleNextPage}
            handlePreviousPage={handlePreviousPage}
            handlePageNumber={handlePageNumber}
          />
          <div style={{ width: "35%" }}>
            <BlueButton htmlType="button" onClick={handleConfirm}>
              CONFIRM AND IMPORT
            </BlueButton>
          </div>
        </div>
      </FormLayout>
      <DialogMessageBox
        showModal={messageBox}
        message={dialogMessage}
        heading={dialogHeading}
        toggleDialog={toggleMessageModal}
        route={STUDENTS}
      />
    </Wrapper>
  ) : (
    <Error statusCode={404} title="Page Not Found" />
  );
};

const mapStateToProps = (store: IStore) => {
  const { university, auth } = store;
  return {
    csvFile: university?.csvFile,
    csvErrors: university?.csvErrors,
    user: auth?.user,
  };
};

const mapDispatchToProps = {
  resetCsv: UniversityPortalActions.ResetCsvData,
};
const ConfirmImport = connect(
  mapStateToProps,
  mapDispatchToProps
)(ConfirmImportComponent);

export { ConfirmImport };
