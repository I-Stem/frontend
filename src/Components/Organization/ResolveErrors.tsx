import React, { useEffect, useRef } from "react";
import Head from "next/head";
import { connect } from "react-redux";
// #endregion Global Imports

// #region Local Imports
import "./style.scss";
import { IStemServices, IStore } from "@Interfaces";
import { Wrapper } from "@Components";
import { FormLayout } from "@Components/HOC/Dashboard/FormHOC";
import { BlueButton } from "@Components/HOC/Dashboard";
import { Table } from "react-bootstrap";
import {
  UPLOAD_EMPLOYEES,
  UPLOAD_STUDENTS,
} from "@Definitions/Constants/universityRoutes";
import { UserType } from "@Definitions/Constants";
import { useAppAbility } from "@Hooks/useAppAbility";
import Error from "next/error";

const ResolveErrorsComponent: React.FC = (props: any) => {
  const initialFocus = useRef<HTMLDivElement>(null);
  const { userType, role, organizationCode } = props.user;
  const { can } = useAppAbility();
  const access = can("VIEW", "STUDENTS");
  const rows = props.csvErrors.validationResult.map((data: any) => {
    return (
      <tr key={data.row}>
        <td>{data.row}</td>
        <td>{data.errors.EMAIL} </td>
        <td>{data.errors.NAME}</td>
        <td>{data.errors.ROLL_NO}</td>
      </tr>
    );
  });
  useEffect(() => {
    initialFocus?.current?.focus();
  });
  return access ? (
    <Wrapper>
      <Head>
        <title>Resolve Errors | I-Stem</title>
      </Head>

      <FormLayout form="resolveErrors" hideFooter={true}>
        <div className="lip-margin">
          <div tabIndex={-1} ref={initialFocus}>
            <h2 className="mt-8 lip-title orange-title">
              RESOLVE THE FOLLOWING ERRORS!
            </h2>
          </div>
          <div className="">
            <Table style={{ marginTop: "1rem" }}>
              <thead>
                <tr style={{ borderTop: "hidden" }}>
                  <th style={{ width: "15%" }}>Row</th>
                  <th>Email</th>
                  <th>
                    {userType === UserType.BUSINESS ? "Employee" : "Student"}{" "}
                    Name
                  </th>
                  <th>Roll Number</th>
                </tr>
              </thead>
              <tbody>{rows}</tbody>
            </Table>
          </div>
          <div style={{ width: "35%", marginTop: "2rem" }}>
            <BlueButton
              href={
                userType === UserType.BUSINESS
                  ? UPLOAD_EMPLOYEES
                  : UPLOAD_STUDENTS
              }
            >
              RE-UPLOAD CSV FILE
            </BlueButton>
          </div>
        </div>
      </FormLayout>
    </Wrapper>
  ) : (
    <Error title="Page Not Found" statusCode={404} />
  );
};

const mapStateToProps = (store: IStore) => {
  const { university, auth } = store;
  return {
    csvFile: university?.csvFile,
    csvErrors: university?.csvErrors,
    user: auth.user,
  };
};

const ResolveErrors = connect(mapStateToProps, null)(ResolveErrorsComponent);
export { ResolveErrors };
