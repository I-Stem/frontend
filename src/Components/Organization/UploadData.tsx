import React, { useEffect, useRef, useState } from "react";
import { NextPage } from "next";
import Head from "next/head";
import { connect } from "react-redux";
import { useRouter } from "next/router";
// #endregion Global Imports

// #region Local Imports
import "./style.scss";
import { IStemServices, IStore } from "@Interfaces";
import { Wrapper } from "@Components";
import { FormLayout } from "@Components/HOC/Dashboard/FormHOC";
import CSVReader from "react-csv-reader";
import { UniversityPortalActions } from "@Actions/UniversityActions";
import {
  COLUMN_MAPPING_EMPLOYEES,
  COLUMN_MAPPING_STUDENTS,
} from "@Definitions/Constants/universityRoutes";
import fileNames from "@Definitions/Constants/image";
import Link from "next/link";
import { UserType } from "@Definitions/Constants";
import { useAppAbility } from "@Hooks/useAppAbility";
import Error from "next/error";

const UploadDataComponent: React.FC = (props: any) => {
  const initialFocus = useRef<HTMLDivElement>(null);
  const [showError, setShowError] = useState(false);
  const { userType } = props.user;
  const router = useRouter();
  useEffect(() => {
    initialFocus?.current?.focus();
  });
  const { can } = useAppAbility();
  const access = can("VIEW", "STUDENTS");
  const handleFileUpload = (data: any, fileInfo: any) => {
    if (data.length) {
      props.uploadCsv({
        csvFile: data,
      });
      if (userType === UserType.BUSINESS) router.push(COLUMN_MAPPING_EMPLOYEES);
      else router.push(COLUMN_MAPPING_STUDENTS);
    } else {
      setShowError(true);
    }
  };
  const parseOptions = {
    header: true,
    dynamicTyping: true,
    skipEmptyLines: true,
    transformHeader: (header: any) => header.toUpperCase().replace(/\W/g, "_"),
  };
  return access ? (
    <Wrapper>
      <Head>
        <title>
          Upload {userType === UserType.BUSINESS ? "Employee" : "Student"} Data
          | I-Stem
        </title>
      </Head>
      <FormLayout form="uploadCSV" hideFooter>
        <div className="lip-margin">
          <div tabIndex={-1} ref={initialFocus}>
            <h2 className="mt-8 lip-title">
              STEP 1: UPLOAD{" "}
              {userType === UserType.BUSINESS ? "EMPLOYEE" : "STUDENT"} DATA
            </h2>
          </div>
          <p className="lip-subtext">
            Upload the CSV file with name, email id and roll no(optional) fields
            for {userType === UserType.BUSINESS ? "employees" : "student"} data.
            Email ID must be unique. Download a sample file
            <Link href={fileNames.IMPORT_STUDENTS_SAMPLE}> here.</Link>
          </p>
          <div className="stud-upload">
            <CSVReader
              cssClass="csv-reader-input"
              label="UPLOAD CSV FILE"
              onFileLoaded={handleFileUpload}
              parserOptions={parseOptions}
            />
          </div>
          {showError ? (
            <div className="error">Uploaded file is empty!</div>
          ) : (
            <></>
          )}
        </div>
      </FormLayout>
    </Wrapper>
  ) : (
    <Error statusCode={404} title="Page Not Found" />
  );
};

const mapStateToProps = (store: IStore) => {
  const { university, auth } = store;
  return {
    csvFile: university?.csvFile,
    user: auth.user,
  };
};

const mapDispatchToProps = {
  uploadCsv: UniversityPortalActions.UploadCsvFile,
};

const UploadData = connect(
  mapStateToProps,
  mapDispatchToProps
)(UploadDataComponent);

export { UploadData };
