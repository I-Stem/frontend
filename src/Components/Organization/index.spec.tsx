import * as React from "react";
import { render } from "@Test/utils";
import { ConfirmImport } from "@Components/Organization/ConfirmAndImport";
import { UploadData } from "@Components/Organization/UploadData";
import { OrganizationDashboard } from "@Components/Organization/Dashboard";
import { ColumnMapping } from "@Components/Organization/ColumnMapping";
import { ResolveErrors } from "@Components/Organization/ResolveErrors";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { UserType } from "@Definitions/Constants";
import { useAppAbility } from "@Hooks/useAppAbility";
import { useRouter } from "next/router";
const mockStore = configureStore([]);

/*
jest.mock("next/router",() => {
  return {
  useRouter:() => {
    return {
      pathname: "/hello/world"
    }
  }
}
});
*/

jest.mock("../../Hooks/useAppAbility", () => {
  return {
    useAppAbility: () => {
      return {
        can: (sub: string, obj: string) => {
          return true;
        },
      };
    },
  };
});

describe("organization pages", () => {
  let store: any;

  beforeEach(() => {
    store = mockStore({
      auth: {
        user: {
          organizationCode: "world",
          userType: UserType.UNIVERSITY,
          role: "STAFF",
        },
      },
      university: {
        csvFile: [["name", "email", "role number"]],
        csvErrors: {
          validationResult: [
            {
              row: 1,
              errors: {
                EMAIL: "Invalid character in domain",
                NAME: "Too long",
                ROLL_NO: "unknown",
              },
            },
          ],
        },
      },
    });
  });

  it("should render without fail", () => {
    const { getAllByRole } = render(
      <Provider store={store}>
        <ConfirmImport />
      </Provider>
    );

    expect(getAllByRole("button")).toBeTruthy();
  });

  it("should match ConfirmAndImport snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <ConfirmImport />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match UploadData snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <UploadData />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  /*
  it("should match Organization dashboard snapshot", () => {
    const { container } = render((
      <Provider store={store}> <OrganizationDashboard/></Provider>
    ));

    expect(container).toMatchSnapshot();
  });
*/
  it("should match ColumnMapping snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        {" "}
        <ColumnMapping />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should match ResolveErrors snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <ResolveErrors />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
