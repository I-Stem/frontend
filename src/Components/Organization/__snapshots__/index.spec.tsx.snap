// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organization pages should match ColumnMapping snapshot 1`] = `
<div>
   
  <section
    class="lipbg"
  >
    <div
      class="ant-row"
    >
      <div
        class="lip-header p-4  w-full flex justify-between"
      >
        <div
          class="PrivateHiddenCss-mdUp-53"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            style="color: rgb(255, 255, 255);"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <img
          alt="I-Stem logo"
          class="lip-logo"
          src="/static/images/logo.svg"
        />
        <button
          class="theme-button hide-border btn btn-primary"
          type="button"
        >
          Accessibility Settings 
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
            style="vertical-align: 0em;"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="text-white float-right pt-1"
        >
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="button"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-center dashboard-container h-auto overflow-auto pb-20"
    >
      <div
        class="ant-col pl-8 ant-col-lg-24"
      >
        <div
          class="lip-margin"
        >
          <div
            tabindex="-1"
          >
            <h2
              class="mt-8 lip-title"
            >
              STEP 2: SELECT RESPECTIVE COLUMN HEADERS
            </h2>
          </div>
          <p
            class="lip-subtext"
          >
            BASIC INFO
          </p>
          <div>
            <form
              class=""
              style="margin-top: 3rem;"
            >
              <div
                class="csv-formgroup form-group"
              >
                <label
                  class="csv-select-label form-label"
                  for="name"
                >
                  <h3
                    class="lip-subtext"
                  >
                    Name
                  </h3>
                </label>
                <select
                  class="stud-select form-control"
                  id="name"
                  name="name"
                  placeholder="Select csv column"
                >
                  <option
                    hidden=""
                    value="none"
                  >
                    Select csv column
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    0
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    1
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    2
                  </option>
                </select>
              </div>
              <div
                class="csv-formgroup form-group"
              >
                <label
                  class="csv-select-label form-label"
                  for="email"
                >
                  <h3
                    class="lip-subtext"
                  >
                    Email
                  </h3>
                </label>
                <select
                  class="stud-select form-control"
                  id="email"
                  name="email"
                  placeholder="Select csv column"
                >
                  <option
                    hidden=""
                    value="none"
                  >
                    Select csv column
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    0
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    1
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    2
                  </option>
                </select>
              </div>
              <div
                class="csv-formgroup form-group"
              >
                <label
                  class="csv-select-label form-label"
                  for="roll_no"
                >
                  <h3
                    class="lip-subtext"
                  >
                    Roll No
                  </h3>
                </label>
                <select
                  class="stud-select form-control"
                  id="roll_no"
                  name="roll_no"
                  placeholder="Select csv column"
                >
                  <option
                    hidden=""
                    value="none"
                  >
                    Select csv column
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    0
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    1
                  </option>
                  <option
                    class="csv-dropdown-item"
                  >
                    2
                  </option>
                </select>
              </div>
              <div
                style="width: 35%;"
              >
                <div
                  class="w-full"
                >
                  <button
                    class="ant-btn ant-btn-block text-center blue-btn"
                    type="submit"
                  >
                    <span
                      class="font-semibold text-lg text-white leading-6"
                    >
                      NEXT
                    </span>
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`organization pages should match ConfirmAndImport snapshot 1`] = `
<div>
  <section
    class="lipbg"
  >
    <div
      class="ant-row"
    >
      <div
        class="lip-header p-4  w-full flex justify-between"
      >
        <div
          class="PrivateHiddenCss-mdUp-23"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            style="color: rgb(255, 255, 255);"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <img
          alt="I-Stem logo"
          class="lip-logo"
          src="/static/images/logo.svg"
        />
        <button
          class="theme-button hide-border btn btn-primary"
          type="button"
        >
          Accessibility Settings 
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
            style="vertical-align: 0em;"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="text-white float-right pt-1"
        >
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="button"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-center dashboard-container h-auto overflow-auto pb-20"
    >
      <div
        class="ant-col pl-8 ant-col-lg-24"
      >
        <div
          class="lip-margin"
        >
          <div
            tabindex="-1"
          >
            <h2
              class="mt-8 lip-title"
            >
              STEP 3: CONFIRM AND IMPORT (
              1
               
              STUDENTS
              )
            </h2>
          </div>
          <div
            class=""
          >
            <table
              class="table"
              style="margin-top: 1rem;"
            >
              <thead>
                <tr
                  style="border-top: hidden;"
                >
                  <th>
                    NAME
                  </th>
                  <th>
                    Email
                  </th>
                  <th>
                    ROLL NUMBER
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td />
                  <td />
                  <td />
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="lip-pagination auto-x-overflow"
          >
            <ul
              class="pagination"
            >
              <li
                class="page-item disabled"
              >
                <span
                  class="page-link"
                  disabled=""
                >
                  <span
                    aria-hidden="true"
                  >
                    Previous
                  </span>
                  <span
                    class="sr-only"
                  >
                    Previous
                  </span>
                </span>
              </li>
              <li
                class="page-item active disabled"
              >
                <span
                  class="page-link"
                  disabled=""
                  value="1"
                >
                  1
                  <span
                    class="sr-only"
                  >
                    (current)
                  </span>
                </span>
              </li>
              <li
                class="page-item disabled"
              >
                <span
                  class="page-link"
                  disabled=""
                >
                  <span
                    aria-hidden="true"
                  >
                    Next
                  </span>
                  <span
                    class="sr-only"
                  >
                    Next
                  </span>
                </span>
              </li>
            </ul>
          </div>
          <div
            style="width: 35%;"
          >
            <div
              class="w-full"
            >
              <button
                class="ant-btn ant-btn-block text-center blue-btn"
                type="button"
              >
                <span
                  class="font-semibold text-lg text-white leading-6"
                >
                  CONFIRM AND IMPORT
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`organization pages should match ResolveErrors snapshot 1`] = `
<div>
  <section
    class="lipbg"
  >
    <div
      class="ant-row"
    >
      <div
        class="lip-header p-4  w-full flex justify-between"
      >
        <div
          class="PrivateHiddenCss-mdUp-68"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            style="color: rgb(255, 255, 255);"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <img
          alt="I-Stem logo"
          class="lip-logo"
          src="/static/images/logo.svg"
        />
        <button
          class="theme-button hide-border btn btn-primary"
          type="button"
        >
          Accessibility Settings 
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
            style="vertical-align: 0em;"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="text-white float-right pt-1"
        >
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="button"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-center dashboard-container h-auto overflow-auto pb-20"
    >
      <div
        class="ant-col pl-8 ant-col-lg-24"
      >
        <div
          class="lip-margin"
        >
          <div
            tabindex="-1"
          >
            <h2
              class="mt-8 lip-title orange-title"
            >
              RESOLVE THE FOLLOWING ERRORS!
            </h2>
          </div>
          <div
            class=""
          >
            <table
              class="table"
              style="margin-top: 1rem;"
            >
              <thead>
                <tr
                  style="border-top: hidden;"
                >
                  <th
                    style="width: 15%;"
                  >
                    Row
                  </th>
                  <th>
                    Email
                  </th>
                  <th>
                    Student
                     
                    Name
                  </th>
                  <th>
                    Roll Number
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    1
                  </td>
                  <td>
                    Invalid character in domain
                     
                  </td>
                  <td>
                    Too long
                  </td>
                  <td>
                    unknown
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            style="width: 35%; margin-top: 2rem;"
          >
            <div
              class="w-full"
            >
              <button
                class="ant-btn ant-btn-block text-center blue-btn"
                type="button"
              >
                <span
                  class="font-semibold text-lg text-white leading-6"
                >
                  RE-UPLOAD CSV FILE
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`organization pages should match UploadData snapshot 1`] = `
<div>
  <section
    class="lipbg"
  >
    <div
      class="ant-row"
    >
      <div
        class="lip-header p-4  w-full flex justify-between"
      >
        <div
          class="PrivateHiddenCss-mdUp-38"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
            style="color: rgb(255, 255, 255);"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <img
          alt="I-Stem logo"
          class="lip-logo"
          src="/static/images/logo.svg"
        />
        <button
          class="theme-button hide-border btn btn-primary"
          type="button"
        >
          Accessibility Settings 
          <span
            aria-label="down"
            class="anticon anticon-down"
            role="img"
            style="vertical-align: 0em;"
          >
            <svg
              aria-hidden="true"
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </button>
        <div
          class="text-white float-right pt-1"
        >
          <span
            aria-label="close"
            class="anticon anticon-close"
            role="button"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="close"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-row-center dashboard-container h-auto overflow-auto pb-20"
    >
      <div
        class="ant-col pl-8 ant-col-lg-24"
      >
        <div
          class="lip-margin"
        >
          <div
            tabindex="-1"
          >
            <h2
              class="mt-8 lip-title"
            >
              STEP 1: UPLOAD
               
              STUDENT
               DATA
            </h2>
          </div>
          <p
            class="lip-subtext"
          >
            Upload the CSV file with name, email id and roll no(optional) fields for 
            student
             data. Email ID must be unique. Download a sample file
            <a
              href="/static/images/IMPORT_STUDENTS_SAMPLE.csv"
            >
               here.
            </a>
          </p>
          <div
            class="stud-upload"
          >
            <div
              class="csv-reader-input"
            >
              <label
                class="csv-label"
                for="react-csv-reader-input"
              >
                UPLOAD CSV FILE
              </label>
              <input
                accept=".csv, text/csv"
                class="csv-input"
                id="react-csv-reader-input"
                name="react-csv-reader-input"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
